packages/react-ui-ag/src/Listings/Listing.js

```diff
@@ -0,0 +1,287 @@
+      } else if (!isActive && this.props.onClick) {
+        this.props.onClick(this.props.index, listing)
+      }
+
+      const shouldStopPropagation =
+        isActive && event && event.stopPropagation
+
+      if (shouldStopPropagation) event.stopPropagation()
+    }
+  }
+
+  @autobind
+  handleFavoriteClick(value, event) {
+    const { isActive, listing, favoriteButton } = this.props
+    const { onClick } = favoriteButton
+
+    if (isActive && onClick) {
+      onClick(this.props.listing, value)
+    } else if (!isActive && this.props.onClick) {
+      this.props.onClick(this.props.index, listing)
+    }
+
+    const shouldStopPropagation = isActive && event && event.stopPropagation
+
+    if (shouldStopPropagation) {
+      event.stopPropagation()
+    }
+  }
+
+  @autobind
+  handlePhotoCarouselSlide(index) { // eslint-disable-line no-unused-vars
+    this.setState({
+      loadedCarousel: true,
+    })
+  }
+
+  renderCtaButtons() {
+    const { ctaButtons } = this.props
+    return ctaButtons.map((cta, index) => this.renderCtaButton(cta, index))
+  }
+
+  renderCtaButton(props, key) {
+    const { theme, listing } = this.props
+    const {
+      className,
+      onClick,
+      valueLocation,
+      children,
+      ...rest
+    } = props
+
+    const buttonText = get(listing, valueLocation, children)
+    const buttonProps = { ...rest }
+
+    if (!buttonText) return null
+
+    return (
+      <Button
+        {...buttonProps}
+        className={classnames(
+          theme.Listing_CtaButton,
+          className,
+        )}
+        onClick={this.handleButtonClick(onClick)}
+        key={key}
+        data-tid="cta-button"
+      >
+        {buttonText}
+      </Button>
+    )
+  }
+
+  renderFavoriteButton() {
+    const { theme, favoriteButton, isActive, listing } = this.props
+    const { className } = favoriteButton
+
+    return (
+      <ToggleButton
+        {...favoriteButton}
+        value={listing.isFavorited}
+        className={classnames(
+          theme.Listing_FavoriteButton,
+          className,
+        )}
+        inactive={!isActive}
+        onClick={this.handleFavoriteClick}
+      />
+    )
+  }
+
+  renderBanners() {
+    const { listing, theme } = this.props
+
+    return (
+      <div className={theme.Listing_Banner}>
+        {listing.banners.map(banner =>
+          (
+            <Banner
+              key={banner}
+              name={banner}
+            />
+          )
+        )}
+      </div>
+    )
+  }
+
+  renderPhotoCarousel() {
+    const { listing, theme, photos, isActive } = this.props
+    let { lazyLoad } = this.props
+
+    if (lazyLoad && typeof lazyLoad === 'boolean') {
+      lazyLoad = typeof lazyLoad === 'boolean'
+        ? REACT_LAZYLOAD_PROP_DEFAULTS
+        : lazyLoad
+    }
+
+    if (!isActive) {
+      const { server, dimensions } = photos
+      const firstPhoto = this.listingPhotos[0] || {}
+      listing.photo = { url: `${server}${firstPhoto.path}${dimensions}` }
```

Comment:
can we use allPhotos function from photos utility(apps/shared/Listings/photos.js)? if we will use that utility we do not need to have photos object listing photo url will be taken care by photos utility like we did in DesktopMapPinListing.js.

Reply:
I'd say this can wait for a future refactor. @tadjohnston ?

Reply:
Yeah, I like the idea @paudyalhari.  We can fix this later as we should probably rethink how we want to do photos in react-ui.